scene/gui/graph_node.cpp

```diff
@@ -503,13 +503,43 @@ bool GraphNode::is_slot_enabled_left(int p_idx) const {
 	return slot_info[p_idx].enable_left;
 }
 
+void GraphNode::set_slot_enabled_left(int p_idx, bool p_enable_left) {
+	ERR_FAIL_COND_MSG(p_idx < 0, vformat("Cannot set enable_left because Slot ID p_idx = %d is smaller than 0.", p_idx));
+
+	slot_info[p_idx].enable_left = p_enable_left;
+	update();
+	connpos_dirty = true;
+
+	emit_signal("slot_updated", p_idx);
+}
+
+void GraphNode::set_slot_type_left(int p_idx, int p_type_left) {
+	ERR_FAIL_COND_MSG(!slot_info.has(p_idx), vformat("Cannot set type_left beacuse slot %d doesn't exist.", p_idx));
+
+	slot_info[p_idx].type_left = p_type_left;
+	update();
+	connpos_dirty = true;
+
+	emit_signal("slot_updated", p_idx);
+}
+
 int GraphNode::get_slot_type_left(int p_idx) const {
 	if (!slot_info.has(p_idx)) {
 		return 0;
 	}
 	return slot_info[p_idx].type_left;
 }
 
+void GraphNode::set_slot_color_left(int p_idx, const Color &p_color_left) {
+	ERR_FAIL_COND_MSG(!slot_info.has(p_idx), vformat("Cannot set color_left beacuse slot %d doesn't exist.", p_idx));
```

Comment:
```suggestion
	ERR_FAIL_COND_MSG(!slot_info.has(p_idx), vformat("Cannot set color_left for a slot '%d' because it hasn't been enabled.", p_idx));
```